internal/resource/registry.go

```diff
@@ -20,6 +20,34 @@ var (
 	kindRegexp         = regexp.MustCompile(`^[A-Z][A-Za-z\d]+$`)
 )
 
+// Scope describes the tenancy scope of a resource.
+type Scope int
+
+const (
+	// There is no default scope, it must be set explicitly.
+	ScopeUndefined Scope = iota
+	// ScopeCluster describes a resource that is scoped to a cluster.
+	ScopeCluster
+	// ScopePartition describes a resource that is scoped to a partition.
+	ScopePartition
+	// ScopeNamespace applies to a resource that is scoped to a partition and namespace.
+	ScopeNamespace
+)
+
+func (s Scope) String() string {
+	switch s {
+	case ScopeUndefined:
+		return "undefined"
+	case ScopeCluster:
+		return "cluster"
+	case ScopePartition:
+		return "partition"
+	case ScopeNamespace:
+		return "namespace"
+	}
+	panic(fmt.Sprintf("string mapping missing for scope %v", int(s)))
```

Comment:
Not sure how I feel about a panic here. Is this _actually_ a fatal error, or can we default to "undefined"?

Reply:
This only happens when someone adds a new enum and forgets to add the string mapping. The system shouldn't be able to run when that is the case since it is required and an obvious development time oversight.